### ISSUE TITLE
Proposal: running containers on multiple nodes with constraint:global

### ISSUE THREAD

**Issue by chanwit** [docker/swarm#1708]
This issue proposes a new `constraint` to use alone or in conjunction with other constraints / affinities to force swarm to run newly created containers on "every" node that meets the criteria.
There are common use cases that users need to run multiple containers at once,
especially when they are scaling services to a specific zone of the cluster. (For example, a swarm cluster spanning across multi-cloud providers).

**Run on every node:**

```
$ docker run -e constraint:global nginx
```

The above command will run nginx onto every node returned by the scheduler.

**Run on some nodes:**

```
$ docker run -e constraint:global -e constraint:region==thailand nginx
```

The above command will run nginx onto every node label as `region==thailand` as they will be returned by the filter.

This is not as elegant as proposed by @aluzzardi in #601. But this constraint will partially implement global scheduling but with different (static) semantic. It will just work and usable in many common cases anyway.

cc @docker/swarm-maintainers

This will also solve #1705 (also ping @winggundamth)

**Comment by vieux** [comment]
I think it's a good idea, I'd also add a new "state" in the node managment, between "pending" and "healthy" like "warming up" so we don't start scheduling containers on a node until all it's global are launched.

The main issue will be docker ps ? do you expect one line per container ?

**Comment by chanwit** [comment]
> The main issue will be docker ps ? do you expect one line per container ?
Similar to a `dangling`, I'm thinking to propose:
```
docker ps --filter grouping=true
```
to allow grouping together all containers started by the same `global` constraint.
So this kind of filter will be available only when using swarm, just like `-e constraint:`.

WDYT?

**Comment by jimmyxian** [comment]
@chanwit Good idea. +1 for this.
When adding new node in cluster, does `global` container need to be auto started on the new node?

**Comment by chanwit** [comment]
To make it deterministic, it won't.
:-)

**Comment by anarcher** [comment]
Good idea. +1

Sometimes, some containers like system/docker metric collector (cadviser, prometheus's exporter..) can be distributed to all nodes or some nodes. 

I think that the proposal is useful for orchestration.

**Comment by MBuffenoir** [comment]
or registrator also, so +1 too.

This way I could even simply add it to my compose instead of my docker-machine provisioning script and by adding some depends_on to be sure it's started before doing anything else ...

**Comment by Typz** [comment]
This can be implemented with existing container constraints:

```
service:
    environment:
      - "affinity:container!=*service*"
```

This works well when starting the app (with `docker-compose up` or `docker-compose scale service=N`), but compose will fail when trying to use `docker-compose up` after images have been updated:

```
$ docker-compose up
creating app_service_1
$ docker-compose scale service=2
creating app_service_2
$ docker-compose build
Sending build context to Docker daemon
[...]
Successfully built a8b6ca53ce
$ docker-compose up
Recreating app_service_1

ERROR: for service  unable to find a node that satisfies container!=*service*
```

**Comment by leecalcote** [comment]
What's different about this proposal than what has already been included in 1.12?

**Comment by winggundamth** [comment]
You can close this since this proposal has been requested since Docker 1.10
